theatre/studio/src/uiComponents/useDrag.ts

```diff
@@ -172,8 +173,18 @@ export default function useDrag(
   }>({onDrag: noop, onDragEnd: noop})
 
   const capturedPointerRef = useRef<undefined | CapturedPointer>()
+  // needed to have a state on the react lifecycle which can be updated
+  // via a ref (e.g. via the below layout effect).
+  const [isDraggingRef, isDragging] = useRefAndState(false)
   useLayoutEffect(() => {
     if (!target) return
+    const ensureIsDraggingUpToDateForReactLifecycle = () => {
+      const isDragging =
+        stateRef.current.domDragStarted && stateRef.current.detection.detected
+      if (isDraggingRef.current !== isDragging) {
+        isDraggingRef.current = isDragging
```

Comment:
Can probably add a comment for to hint for a future refactor:
```suggestion
       // TODO move all operaitons on isDraggingRef to a function that automatically updates the isDragging state as well
        isDraggingRef.current = isDragging
```